main.go

```diff
@@ -77,12 +77,12 @@ func listInternalSources(limit, offset int64) *SourceResponse {
 	log.Printf("Requesting [limit %v] + [offset %v] sources from internal API at [%v]", limit, offset, host)
 
 	url, _ := url.Parse(fmt.Sprintf("%v/internal/v2.0/sources?limit=%v&offset=%v", host, limit, offset))
-	req := &http.Request{Method: "GET", URL: url, Header: map[string][]string{
+	req := &http.Request{Method: http.MethodGet, URL: url, Header: map[string][]string{
 		"x-rh-sources-account-number": {"sources_monitor"},
 		"x-rh-sources-psk":            {psk},
 	}}
 	resp, err := httpClient.Do(req)
-	if err != nil || resp.StatusCode != 200 {
+	if err != nil || (resp != nil && resp.StatusCode != 200) {
```

Comment:
Also, since we're refactoring this bit, I would suggest using `http`'s defined constants for the status codes:

```diff
- if err != nil || resp.StatusCode != 200 {
+ if err != nil || resp.StatusCode != http.StatusOK {
```